pkg/runs/runsReset.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package runs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/galasa-dev/cli/pkg/embedded"
+	galasaErrors "github.com/galasa-dev/cli/pkg/errors"
+	galasaapi "github.com/galasa-dev/cli/pkg/galasaapi"
+	"github.com/galasa-dev/cli/pkg/utils"
+)
+
+var (
+	RESET_STATUS = "queued"
+	RESET_RESULT = ""
+)
+
+func ResetRun(
+	runName string,
+	timeService utils.TimeService,
+	console utils.Console,
+	apiServerUrl string,
+	apiClient *galasaapi.APIClient,
+) error {
+	var err error
+	var runId string
+
+	log.Printf("ResetRun entered.")
+
+	if runName == "" {
+		err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_MISSING_NAME_FLAG, runName)
+	}
+
+	if (err == nil) && (runName != "") {
+		err = ValidateRunName(runName)
+	}
+
+	if err == nil {
+
+		runId, err = getRunIdFromRunName(runName, timeService, apiClient)
+
+		if err == nil {
+
+			updateRunStatusRequest := createUpdateRunStatusRequest(RESET_STATUS, RESET_RESULT)
+
+			err = resetRun(runName, runId, updateRunStatusRequest, apiClient)
+
+			if err == nil {
+				consoleErr := console.WriteString(fmt.Sprintf(galasaErrors.GALASA_INFO_RUNS_RESET_SUCCESS.Template, runName))
+
+				// Console error is not as important to report as the original error if there was one.
+				if consoleErr != nil && err == nil {
+					err = consoleErr
+				}
+			}
+
+		}
+
+	}
+
+	log.Printf("ResetRun exiting. err is %v", err)
```

Comment:
'\n' needed at the end of the printf ?
Else the text will run into the following printf text.
Not sure it does that for log, but it will for most other output streams.

Reply:
Added new line character in case.